vsphere_plugin_common/__init__.py

```diff
@@ -1664,14 +1718,23 @@ def create_server(
         relospec.datastore = datastore.obj
         relospec.pool = resource_pool.obj
         if not auto_placement:
-            logger().warn(
+            self._logger.warn(
                 'Disabled autoplacement is not recomended for a cluster.'
             )
             relospec.host = host.obj
 
+        # get list nic mac's for remove
+        macs_for_remove = []
+        if postpone_delete_networks and not enable_start_vm:
+            macs_for_remove = self._get_nic_keys_for_remove(template_vm)
```

Comment:
This isn't the best named variable ever, since as pointed out earlier we can't remove based on MAC. Probably better to name it keys_for_removal?

Reply:
fixed